### PR TITLE
“#1-4Task. margo.astreyko”

### PR DESCRIPTION
**Pull request by Honey-badger1**
https://travis-ci.org/github/AisBrestEDU/js-assignments

### CODE REVIEW

**Comment by IlyaMokin** [comment]
Tests in the comment are failed. Please put a link to travic-ci build under your account. Also do not hesitate to change version of nodejs in travis config if needed.

**Comment by Honey-badger1** [comment]
Updated and done. https://travis-ci.org/github/Honey-badger1/js-assignments

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function concatenateStrings(value1, value2) {
  *   ''      => 0
  */
 function getStringLength(value) {
-    throw new Error('Not implemented');
+    return (value.length);
```
Comment (AlexMokin) [inline task/01-strings-tasks.js:41]: remove redundant brackets. In this and other methods

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -267,7 +278,14 @@ function isString(value) {
...
  */
 function getCardId(value) {
-    throw new Error('Not implemented');
+    let a = ['A♣', '2♣', '3♣', '4♣', '5♣', '6♣', '7♣', '8♣', '9♣', '10♣', 'J♣', 'Q♣', 'K♣',
+        'A♦', '2♦', '3♦', '4♦', '5♦', '6♦', '7♦', '8♦', '9♦', '10♦', 'J♦', 'Q♦', 'K♦',
+        'A♥', '2♥', '3♥', '4♥', '5♥', '6♥', '7♥', '8♥', '9♥', '10♥', 'J♥', 'Q♥', 'K♥',
+        'A♠', '2♠', '3♠', '4♠', '5♠', '6♠', '7♠', '8♠', '9♠', '10♠', 'J♠', 'Q♠', 'K♠'
+    ];
+
+    a.join('');
```
Comment (AlexMokin) [inline task/01-strings-tasks.js:287]: what's this for?

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -159,8 +159,8 @@ function parseNumberFromString(value) {
  *   3,3,3   => 5.196152422706632
  *   1,2,3   => 3.741657386773941
  */
-function getParallelipidedDiagonal(a,b,c) {
-    throw new Error('Not implemented');
+function getParallelipidedDiagonal(a, b, c) {
+    return Math.sqrt(a * a + b * b + c * c);
```
Comment (AlexMokin) [inline task/02-numbers-tasks.js:163]: try to use math.hypot

File: task/03-date-tasks.js [hunk 4]
```diff
@@ -94,14 +109,21 @@ function timeSpanToString(startDate, endDate) {
...
+   if (angle > 180) {
+      return (360 - angle) * (Math.PI / 180);
+   }
+   return (angle * (Math.PI / 180));
+
 }
 
 
 module.exports = {
-    parseDataFromRfc2822: parseDataFromRfc2822,
```
Comment (AlexMokin) [inline task/03-date-tasks.js:102]: keep original formatting for exsisting code

File: task/04-arrays-tasks.js [hunk 5]
```diff
@@ -85,7 +88,12 @@ function getArrayOfPositives(arr) {
  *    [ 'cat, 'dog', 'raccon' ] => [ 'cat', 'dog', 'racoon' ]
  */
 function getArrayOfStrings(arr) {
-   throw new Error('Not implemented');
+   return arr.filter(i => {
+      if (typeof i === 'string') {
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js:92]: just return boolean expression `typeof i === 'string'`

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -559,40 +603,48 @@ function getElementByIndexes(arr, indexes) {
...
+      return result.concat(arr.slice(length / 2 + 1, length), arr[parseInt(length / 2, 10)], arr.slice(0, length / 2));
+   }
 }
 
 
+
 module.exports = {
-    findElement: findElement,
-    generateOdds: generateOdds,
-    doubleArray: doubleArray,
```
Comment (AlexMokin) [inline task/04-arrays-tasks.js:569]: keep original formatting for the original code

**Comment by AlexMokin** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IlyaMokin** [comment]
Thank you! Closing as accepted!

**Pull request closed without merging**